### DISCUSSION TITLE
Exclude visitors by IP address

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#112] [Anything Else]
"It would be great to have an ignore list for IP addresses. Currently, my and my team’s manual testing shows up as visits.

Dunno if this is any help but if you’re testing on any environments that aren’t “production” then wrapping the Plausible script tag inside a simple IF statement does the trick. EG I’m rarely on production but I’m messing about in LOCAL all the time so my Plausible script is inside

if (environment == “Production”)
// script tag
}

so nothing from local, dev or staging gets counted.

This would only be for client-side events but we use Google server-side for event tracking. Additionally, the method above would require getting your employees to install it on their phones and computers which seems a bit unrealistic.

Many start also need to block the tracking of brower testing software (ex: Browerstack - https://www.browserstack.com/question/585) and security scanners."

**Update**: It’s now easy to exclude your internal traffic from being counted in your stats. See how here: https://plausible.io/docs/excluding

**Comment by ghost** [comment]
Temporary workaround: [https://docs.plausible.io/excluding/](https://docs.plausible.io/excluding/)

**Reply by Alpine418** [reply]
Not an option if your company does not allow you to install any 3rd party browser extensions.

**Reply by thexpand** [reply]
Also, not an option if you're visiting from a mobile device.

**Comment by metmarkosaric** [comment]
You can now exclude your own views by localstorage too. See https://plausible.io/docs/excluding

**Comment by metmarkosaric** [comment]
"We need a simple way for our employees and testers to opt out on mobile devices to not dirty the analytics up"

(feedback via email)

**Reply by Vigasaurus** [reply]
I'd say this is easily possible with the localStorage method - the site owner/developer would just need to expose a way for the site to set the localStorage flag without a user having to do it themselves in the console (to make it mobile/non-technical friendly). 

Just a simple page on a semi-private route with a button to toggle the localStorage flag on and off could do the trick. If you want, I can write up a super basic version of this for the docs, that people could reuse as they'd like.

**Reply by metmarkosaric** [reply]
Yeah, that would be great @Vigasaurus, thanks!

**Reply by ukutaht** [reply]
ahh good point. @metmarkosaric when you asked me yesterday I didn't think of this sort of solution. Could work very well.

**Reply by arladmin** [reply]
Hi all. Kindly advise how to implement this in the case of simple one-page websites (no possibility of additional routes/private logins). Also, majority of our traffic is from mobile browsers. How to implement the localStorage method in such cases?

**Reply by Vigasaurus** [reply]
I'd say the same general method can still work - you basically just put a button/checkbox somewhere on the page (of course it being on the public page isn't ideal, but it can work), that can toggle the localStorage flag.

**Reply by metmarkosaric** [reply]
Just got a similar request via email: "We are looking forward to trying out your platform. We currently exclude all of our internal website traffic from our Google Analytics, I’ve found in your in documentation this is possible via a console command but we would need to set this on all of our computers (500+). Are you aware of a setting within group policy that would allow us to set this flag across our estate?"

**Reply by Vigasaurus** [reply]
I'll make the docs for that today 👍

**Reply by metmarkosaric** [reply]
thank you!

**Reply by metmarkosaric** [reply]
docs are now updated thanks to @Vigasaurus! 

see https://plausible.io/docs/excluding#allow-anyone-on-your-site-to-exclude-themselves

**Comment by metmarkosaric** [comment]
Some feedback on: https://plausible.io/docs/excluding#allow-anyone-on-your-site-to-exclude-themselves

"I think that could work for us, I’d prefer to have no user intervention so the item will be created on page load:

```
<script>

    window.onload = loadFunction;

  function loadFunction() {

        if (window.localStorage.getItem("isNewSession")) {

        } else {

            window.localStorage.setItem('isNewSession', 'true');

            window.localStorage.setItem('plausible_ignore','true');

        }

}

</script>
```

I will then set this script to only run for our internal users."

What do you think @Vigasaurus? Something that could be added to the docs for others to replicate?

**Reply by Vigasaurus** [reply]
Yeah that could be added, I figured most people would want it as an option though. Maybe I can add it in as a commented line for those who want it to auto set on load (and we can change the default state of that as more feedback dictates). 

Realistically though, I think having the current example available is plenty, as most users who could/would read that part of the docs are more than capable of making a page to do what the example does in no time at all (or make any changes they'd like) - so long as the examples/API is made clear (I'd say it is, already). But at the same time, having more available as needed can't hurt.

**Reply by metmarkosaric** [reply]
Yeah, optional for those that don't want their testers to take any action whatsoever.

**Reply by MattEhrlich** [reply]
I was just looking for a solution to this - Here's what I did if it helps anyone.
```
          window.addEventListener('load', () => {
            var exclusionState = window.localStorage.plausible_ignore == "true"

            if (exclusionState) {
              console.log('We are excluding your visits from marketing analytics');
            } else {
              window.localStorage.plausible_ignore = "true";
            }
          });
```
This is inside our **Admin landing page on log in** so once an admin has logged in, we auto-tag them to block their visits. The flag will stay even on log out and only be removed if they clear local storage. At which point, once they log back in. They'll be retagged to block from analytics. 

This is the easiest way I can think of that doesn't require every site admin to click a button on all browsers on all devices. 

Thoughts?

**Reply by ukutaht** [reply]
yeah, that seems like a perfectly good solution to me

**Comment by Alpine418** [comment]
I am planning to add support for visitor IP exclusion in the Plausible Analytics Reverse Proxy.

https://github.com/Neoflow/ReverseProxy-PlausibleAnalytics

Anybody intersted to use it?

**Reply by Alpine418** [reply]
Done :) Please give a feedback if you use it or try it out. Thanks!

https://github.com/Neoflow/ReverseProxy-PlausibleAnalytics/releases/tag/1.2.0

**Comment by metmarkosaric** [comment]
**Update**: It’s now easy to exclude your internal traffic from being counted in your stats. See how here: https://plausible.io/docs/excluding
